src/cmd/flux-mini.py

```diff
@@ -0,0 +1,305 @@
+        unit = m.group(1) if m else "s"
+
+        if unit == "m":
+            seconds = timedelta(minutes=n).total_seconds()
+        elif unit == "h":
+            seconds = timedelta(hours=n).total_seconds()
+        elif unit == "d":
+            seconds = timedelta(days=n).total_seconds()
+        else:
+            seconds = n
+        if seconds < 0 or math.isnan(seconds) or math.isinf(seconds):
+            raise ValueError("invalid Flux standard duration")
+        return seconds
+
+    def set_duration(self, duration):
+        """
+        Assign a time limit to the job.  The duration may be:
+        - a float in seconds
+        - a string in Flux Standard Duration
+        A duration of zero is interpreted as "not set".
+        """
+        if type(duration) is str:
+            t = self.__parse_fsd(duration)
+        elif type(duration) is float:
+            t = duration
+        else:
+            raise ValueError("duration must be a float or string")
+        if t < 0:
+            raise ValueError("duration must not be negative")
+        if math.isnan(t) or math.isinf(t):
+            raise ValueError("duration must be a normal, finite value")
+        self.jobspec["attributes"]["system"]["duration"] = t
+
+    def set_cwd(self, cwd):
+        """
+        Set working directory of job.
+        """
+        if type(cwd) is not str:
+            raise ValueError("cwd must be a string")
+        self.jobspec["attributes"]["system"]["cwd"] = cwd
+
+    def set_environment(self, environ):
+        """
+        Set (entire) environment of job.
+        """
+        if type(environ) is not dict:
+            raise ValueError("environment must be a dictionary")
+        self.jobspec["attributes"]["system"]["environment"] = environ
+
+    def set_shell_options(self, opts):
+        """
+        Set (all) Flux shell options for job.
+        """
+        if type(opts) is not dict:
+            raise ValueError("shell options must be a dictionary")
+        self.jobspec["attributes"]["system"]["shell"]["options"] = opts
+
+    def dumps(self):
+        return json.dumps(self.jobspec)
+
+
+class SubmitCmd:
+    """
+    SubmitCmd submits a job, displays the jobid on stdout, and returns.
+
+    Usage: flux mini submit [OPTIONS] cmd ...
+    """
+
+    def __init__(self):
+        self.parser = self.create_parser()
+
+    def create_parser(self):
+        """
+        Create parser with args for submit subcommand
+        """
+        parser = argparse.ArgumentParser(add_help=False)
+        parser.add_argument(
+            "-N", "--nodes", type=int, help="Number of nodes to allocate"
+        )
+        parser.add_argument(
+            "-n", "--ntasks", type=int, default=1, help="Number of tasks to start"
+        )
+        parser.add_argument(
+            "-c",
+            "--cpus-per-task",
+            type=int,
+            default=1,
+            help="Number of cpus to allocate per task",
```

Comment:
Totally optional: Use `metavar='N'` in these `add_argument` calls to get more traditional help output instead of e.g. `-c CPUS_PER_TASK, --cpus-per-task CPUS_PER_TASK`